src/main/java/tb/common/block/BlockRainbowCactus.java

```diff
@@ -41,13 +43,26 @@ public ArrayList<ItemStack> getDrops(World world, int x, int y, int z, int metad
             return ret;
         }
 
-        for (int i = 0; i < 3 + world.rand.nextInt(8); ++i)
-            ret.add(new ItemStack(Items.dye, 1, allowedDyes[world.rand.nextInt(allowedDyes.length)]));
-
+        for (int i = 0; i < 3 + world.rand.nextInt(8); ++i) {
+            if (Loader.isModLoaded("gregtech")) {
+                if (colors == null) colors = addGTColors();
+                ret.add(colors[world.rand.nextInt(colors.length)]);
+            } else ret.add(new ItemStack(Items.dye, 1, allowedDyes[world.rand.nextInt(allowedDyes.length)]));
+        }
```

Comment:
I don't like checking and populating array (`colors`) at runtime. I'd better refactored this code.
Remove `colors` variable, convert `allowedColors` to `ItemStack` array. Then populate this array at init stage with either vanilla or GT dyes. And at runtime just get dyes from this array (no need for additional generation logic).